docs/VERSIONS.md

```diff
@@ -1 +1,51 @@
 # Historical Versions
+
+## tag 0.2.1 (testnet)
+
+commit hash: 53bfd4d4bbbee773ca10c1181982d9a27718826d
+
+**spore**:
+|   |   |
+|---|---|
+|tx_hash|0x06995b9fc19461a2bf9933e57b69af47a20bf0a5bc6c0ffcb85567a2c733f0a1|
+|index|0|
+|code_hash|0x5e063b4c0e7abeaa6a428df3b693521a3050934cf3b0ae97a800d1bc31449398|
+
+**cluster**:
+|   |   |
+|---|---|
+|tx_hash|0xfbceb70b2e683ef3a97865bb88e082e3e5366ee195a9c826e3c07d1026792fcd|
+|index|0|
+|code_hash|0x7366a61534fa7c7e6225ecc0d828ea3b5366adec2b58206f2ee84995fe030075|
+
+**cluster_agent**:
+|   |   |
+|---|---|
+|tx_hash|0x53fdb9366637434ff685d0aca5e2a68a859b6fcaa4b608a7ecca0713fed0f5b7|
+|index|0|
+|code_hash|0xc986099b41d79ca1b2a56ce5874bcda8175440a17298ea5e2bbc3897736b8c21|
+
+**cluster_proxy**:
+|   |   |
+|---|---|
+|tx_hash|0x0231ea581bbc38965e10a2659da326ae840c038a9d0d6849f458b51d94870104|
+|index|0|
+|code_hash|0xbe8b9ce3d05a32c4bb26fe71cd5fc1407ce91e3a8b9e8719be2ab072cef1454b|
+
+## tag 0.2.1 (mainnet)
+
+commit hash: 40ef57a2d3689740bf0c6fa47938e52948252179
```

Comment:
Why is the commit of https://github.com/sporeprotocol/spore-contract/releases/tag/0.2.1 showing [53bfd4d4bbbee773ca10c1181982d9a27718826d](https://github.com/sporeprotocol/spore-contract/commit/53bfd4d4bbbee773ca10c1181982d9a27718826d)?

NOT 40ef57a2d3689740bf0c6fa47938e52948252179

<img width="842" alt="image" src="https://github.com/sporeprotocol/spore-contract/assets/1297478/c10bf016-4bab-4a3c-8eb9-296004450f27">

Reply:
because tag 0.2.1 created in a previous commit, and all of new commits will change repo's commit hash above then

Reply:
"Do NOT change a tag on a remote repository, unless sensitive data was leaked."

Reply:
### See also:
`git help tag`
-> Read the DISCUSSION - On Re-tagging

Reply:
so what we gonna to do? or you can just do it by yourself, I have no idea for this.

Reply:
I am checking the diff between [53bfd4d4bbbee773ca10c1181982d9a27718826d](https://github.com/sporeprotocol/spore-contract/commit/53bfd4d4bbbee773ca10c1181982d9a27718826d) and 40ef57a2d3689740bf0c6fa47938e52948252179.

will give a suggestion later.

Reply:
Can't use [53bfd4d4bbbee773ca10c1181982d9a27718826d](https://github.com/sporeprotocol/spore-contract/commit/53bfd4d4bbbee773ca10c1181982d9a27718826d) to build:

```bash
capsule build --release -- --features release_export

  thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { inner: Error { inner: TomlError { message: "missing field `commit_hash`", original: Some("[[code_hash_list]]\n"), keys: ["code_hash_list"], span: Some(0..18) } } }', lib/build/src/lib.rs:111:29
```

I understood why you use https://github.com/sporeprotocol/spore-contract/commit/40ef57a2d3689740bf0c6fa47938e52948252179 to build.

## A simple solution
just create a new tag 0.2.2 for the mainnet release.